services/util/array-utils/array.utils.go

```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2019 Snowflake Inc. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+*/
+
+package array_utils
+
+// IsInArray checks if value is in array
+func IsInArray[T comparable](arr []T, value T) bool {
+	for _, v := range arr {
+		if v == value {
+			return true
+		}
+	}
+	return false
+}
+
+// FindIndexBy finds the index of value in array
+func FindIndexBy[T comparable](arr []T, predicate func(val T) bool) int {
```

Comment:
I'd try to avoid having `utils` packages. This function is only used in one place. Should we just place this in the same file, to simplify?

Reply:
How I see this 2 options. I'm try to avoid compressing and relay only on absolute facts

If we keep util
*Pros*
- code in a separate module are easy to read
- it logically placed in `array-utils` to simplify finding it
- function covered by test and we sure it have no bugs
- it generic and ready to be reused in other cases without modifications

*Cons*
- extra module

If we not extract it:
*Pros*
- no extra module

*Cons*
- next time when similar usecase appears, it would be required to remember that there was first usecase and refactor it to use generic function
- source module code would complicated to understand and module would look massy, new contributor would need first understand what function is belongs to functionality and what to utils and only after that make required changes

------
Besides I'm not see a big difference would the function be in separate module or in the same one.
I now that a little bit tricky, and could lead to extracting every single pice of code to separate module. But if we keep utils application/domain agnostic everything should be ok. Simple test is that it should be easy to copy function as it is any other app and start use it in 2-3 different usecases. For this one is a ton of search usecases were required custom/complicated logic